src/Evolution/DiscontinuousGalerkin/InboxTags.hpp

```diff
@@ -0,0 +1,117 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <boost/functional/hash.hpp>
+#include <cstddef>
+#include <map>
+#include <optional>
+#include <tuple>
+#include <type_traits>
+#include <utility>
+
+#include "DataStructures/FixedHashMap.hpp"
+#include "Domain/Structure/Direction.hpp"
+#include "Domain/Structure/ElementId.hpp"
+#include "Domain/Structure/MaxNumberOfNeighbors.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "Parallel/InboxInserters.hpp"
+#include "Time/TimeStepId.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace evolution::dg::Tags {
+/*!
+ * \brief The inbox tag for boundary correction communication and DG-subcell
+ * ghost zone cells.
+ *
+ * The first element in the tuple is the mesh of the neighboring element's
+ * face (not the mortar mesh!), the second is the volume variables for finite
+ * difference/volume reconstruction, and the third is the data on the mortar
+ * needed for computing the boundary corrections (e.g. fluxes, characteristic
+ * speeds, conserved variables). The fourth element of the tuple is the range
+ * of validity of the boundary terms. This fourth element is needed when using
+ * local time stepping so that the element knows whether or not it should
+ * remove boundary data and expect new data to be sent from a neighbor. The
+ * TimeStepId also coincides with the next time step of the neighboring
+ * element.
+ *
+ * The ghost cell data (second element of the tuple) will be valid whenever
+ * DG-subcell is being used. Elements using DG and not FD/FV always send both
+ * the ghost cells and boundary correction data togther. Elements using FD/FV
+ * send the ghost cells first followed by the boundary correction data once the
+ * element has received all neighbor ghost cell data.
```

Comment:
Consider documenting (here? below?) that only the flux and timestepid are read from the second send/receive?